src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```
@@ -0,0 +1,114 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        String query = "INSERT INTO products (product_name, product_price)"
+                + " VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, product.getName());
+            preparedStatement.setDouble(2, product.getPrice());
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            throw new RuntimeException("Incorrect create query", e);
```

From checklist: e.printStackTrace() - is a bad practice! Let’s create custom exception DataProcessingException with two parameters: String message and Throwable ex. It should be extended from RuntimeException. You should rethrow this exception in catch black on dao layer.